x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_list.tsx

```diff
@@ -29,41 +34,51 @@ import {
 import { useTrustedAppsSelector } from './hooks';
 
 import { FormattedDate } from '../../../../common/components/formatted_date';
-import { OS_TITLES } from './constants';
-
-const COLUMN_TITLES: Readonly<
-  { [K in keyof Omit<TrustedApp, 'id' | 'description' | 'entries'> | 'actions']: string }
-> = {
-  name: i18n.translate('xpack.securitySolution.trustedapps.list.columns.name', {
-    defaultMessage: 'Name',
-  }),
-  os: i18n.translate('xpack.securitySolution.trustedapps.list.columns.os', {
-    defaultMessage: 'OS',
-  }),
-  created_at: i18n.translate('xpack.securitySolution.trustedapps.list.columns.createdAt', {
-    defaultMessage: 'Date Created',
-  }),
-  created_by: i18n.translate('xpack.securitySolution.trustedapps.list.columns.createdBy', {
-    defaultMessage: 'Created By',
-  }),
-  actions: i18n.translate('xpack.securitySolution.trustedapps.list.columns.actions', {
-    defaultMessage: 'Actions',
-  }),
-};
+import { ACTIONS_COLUMN_TITLE, LIST_ACTIONS, OS_TITLES, PROPERTY_TITLES } from './translations';
+import { TrustedAppCard } from './components/trusted_app_card';
+
+interface DetailsMap {
+  [K: string]: ReactNode;
+}
+
+interface TrustedAppsListContext {
+  dispatch: Dispatch<Immutable<AppAction>>;
+  detailsMapState: [DetailsMap, (value: DetailsMap) => void];
+}
 
+type ColumnsList = Array<EuiBasicTableColumn<Immutable<TrustedApp>>>;
 type ActionsList = EuiTableActionsColumnType<Immutable<TrustedApp>>['actions'];
 
-const getActionDefinitions = (dispatch: Dispatch<Immutable<AppAction>>): ActionsList => [
+const toggleItemDetailsInMap = (
+  map: DetailsMap,
+  item: Immutable<TrustedApp>,
+  { dispatch }: TrustedAppsListContext
+): DetailsMap => {
+  const changedMap = { ...map };
+
+  if (changedMap[item.id]) {
+    delete changedMap[item.id];
+  } else {
+    changedMap[item.id] = (
+      <TrustedAppCard
+        trustedApp={item}
+        onDelete={() => {
```

Comment:
is there a way you can memoize this callback?

Reply:
Will do that as part of my next story, somehow slipped through.